### DISCUSSION TITLE
SeekFwd as secondary button function?

### DISCUSSION THREAD

**Discussion by vizete** [MiczFlor/RPi-Jukebox-RFID#2365] [Q&A] [answered]
Hello,

I've a Jukebox with 5 buttons:

1. VolumeUp
2. VolumeDown
3. PrevSong
4. Play/Pause
5. NextSong

I'm looking for a way to add the functions SeekFwd/SeekBack repeatedly.

I tried a TwoButtonConfig combining VolumeUp and NextSong with a two button call triggering SeekFwd like this:

```
[VolumeUpNextForward]
enabled: True
Type: TwoButtonControl
Pin1: 27
Pin2: 13
pull_up_down: pull_up
hold_time: 0.3
hold_mode: SecondFuncRepeat
functionCall1: functionCallVolU
functionCall2: functionCallPlayerNext
functionCallTwoButtons: functionCallPlayerSeekFwd
;functionCallTwoButtonsArgs: 30
```
But each time I use the two button config for seeking forward I also trigger 'NextSong'.
I also tried to use a RFID card to Seek Fwd but this triggers the action only once.
I want to seek forward as long as the buttons are pressed/the card is presented to the RFID reader. Any ideas how to do this?

This is my setup:
```
System
OS DistributionRaspbian
DescriptionRaspbian GNU/Linux 10 (buster)
Release10
Codenamebuster
Throttling OK
Temperature 40.8'C
Phoniebox Setup
Version2.7.0 - 550a2583 - master
EditionClassic edition (barebones)
```

Many thanks!

**Comment by AlvinSchiller** [comment, marked as answer]
Hi, you can try out the workaround or hack described here: https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1666

**Reply by AlvinSchiller** [reply]
And continuous function calls (e.g. seeking) via RFID card will not be possible without custom implementation. Scanning a card only triggers the action once (the place_not_swipe feature only checks if the card is still present, otherwise pauses the playback).

**Reply by AlvinSchiller** [reply]
This is now fixed with https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2367 for v2.8.0.
The behavior of `hold_mode` `SecondFunc` and `SecondFuncRepeat` has changed. The secondary function is no longer triggered additionally to the primary function.
Now its called exclusively if `hold_time` is reached. The primary function will only be triggered if the button is pressed shorter then `hold_time`! Existing configurations may need to adapt to this.

If you still encounter the execution of `functioncall ` after the hold, try setting the `antibouncehack` to `True`.

**Reply by vizete** [reply]
I gave the hack a chance and it works for me (still on version 2.7.0)
This is my button config for Next/FastForward just in case some is looking for the same problem
```
[NextSongFastForward]
enabled: True
Type: Button
Pin: 13
pull_up_down: pull_up
hold_time: 0.3
hold_mode: SecondFuncRepeat
functionCall: functionCallPlayerNext
functionCall2: functionCallPlayerSeekFwd
;functionCall2Args: 30
antibouncehack: True
```
I will try version 2.8 as soon as I have a little more time/less time pressure.

Many thanks you for your time and effort.

**Reply by s-martin** [reply]
> I will try version 2.8 as soon as I have a little more time/less time pressure.

Attention: 2.8 is not released yet. You would have to install from `develop` branch.
